.github/workflows/test.yml

```diff
@@ -0,0 +1,34 @@
+name: test
+
+on:
+  pull_request: {}
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  RUST_BACKTRACE: short
+  RUSTUP_MAX_RETRIES: 10
+
+permissions:
+  contents: read
+
+jobs:
+
+  # Run all non-integration tests.
+  unit:
+    timeout-minutes: 15
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579
+      - run: cargo test --all --exclude=linkerd-app-integration --no-run
+      - run: cargo test --all --exclude=linkerd-app-integration
```

Comment:
is the idea behind using `--no-run` first that there's a separate step that builds the tests, so we can differentiate between build failures and test failures?

Reply:
yep, recommended in matklad's post. it also helps us be clear about where we're spending time (in ci logs)